lib/src/testing_library/dom/within.dart

```diff
@@ -0,0 +1,30 @@
+import 'dart:html' show document, Element;
+
+import 'package:over_react/over_react.dart';
+import 'package:over_react_test/src/testing_library/dom/queries.dart' show DomQueries;
+import 'package:test/test.dart' show addTearDown, expect, isNotNull, isTrue;
+
+/// Queries scoped to the provided `element`.
+class _WithinQueries extends DomQueries {
+  _WithinQueries(this.element) : super(() => element);
+
+  final Element element;
+}
+
+/// Takes a DOM [element] and binds it to the raw query functions,
+/// allowing them to be used without specifying a container.
+///
+/// > See: <https://testing-library.com/docs/dom-testing-library/api-within>
+_WithinQueries within(Element element) {
```

Comment:
This is how I went about mimicking how the JS library "binds" all the "raw" queries to a specified `Element`.

Reply:
So, building on the example of an overlay / using a screen query - the `within()` "scope" of queries would make it easy to test an "overlay" component's `container` argument... 

```dart
import 'dart:html';

import 'package:over_react/over_react.dart';
import 'package:over_react_test/react_testing_library.dart' as rtl;
import 'overlay_trigger.dart';

test('my component renders an overlay when clicked', () async {
  final customDialogContainer = document.body.append(DivElement()..attributes = {'data-testid': 'customContainer'});
  rtl.render(
    (OverlayTrigger()
      ..overlay = (Dom.div()..role = 'dialog')('I am the overlay!')
      ..container = customDialogContainer
    )(Dom.button()('Click me'))
  );
  
  queryByRole('button').click();
  // This async query will even wait for the overlay to become visible automatically!
  final dialogElem = await within(customDialogContainer).findByRole('dialog');
  expect(dialogElem.text, 'I am the overlay');
});
```